pkg/cli/cmd/radinit/recipe.go

```diff
@@ -52,148 +127,57 @@ func NewDevRecipeClient() DevRecipeClient {
 	tag := version.Channel()
 	if version.IsEdgeChannel() {
 		tag = "latest"
 	}
 
-	// Temporary solution to get all repositories.
-	// The issue is that if RepositoryListPageSize is not specified the default is 100.
-	// We have 104 repositories in the registry as of 12 Oct 2023. That is why processRepositories
-	// function was being called twice and the second call was overwriting all the recipes.
-	// TODO: Remove this once we have a better solution.
-	reg.RepositoryListPageSize = 1000
-
-	recipes := map[string]map[string]corerp.RecipePropertiesClassification{}
-
-	// if repository has the correct path it should look like: <registryPath>/recipes/<category>/<type>:<tag>
-	// Ex: ghcr.io/radius-project/recipes/local-dev/rediscaches:0.20
-	// The start parameter is set to "radius-rp" because our recipes are after that repository.
-	err = reg.Repositories(ctx, "", func(repos []string) error {
-		// validRepos will contain the repositories that have the requested tag.
-		validRepos := []string{}
-		for _, repo := range repos {
-			r, err := reg.Repository(ctx, repo)
-			if err != nil {
-				continue
-			}
-
-			tagExists := false
-			err = r.Tags(ctx, "", func(tags []string) error {
-				for _, t := range tags {
-					if t == tag {
-						tagExists = true
-						break
-					}
-				}
-				return nil
-			})
-			if err != nil {
-				continue
-			}
-
-			if tagExists {
-				validRepos = append(validRepos, repo)
-			}
-		}
-
-		recipes = processRepositories(validRepos, tag)
-		return nil
-	})
-	if err != nil {
-		return nil, fmt.Errorf("failed to list recipes available in registry at path  %s -  %s", DevRecipesRegistry, err.Error())
-	}
-
-	return recipes, nil
-}
-
-// processRepositories processes the repositories and returns the recipes.
-func processRepositories(repos []string, tag string) map[string]map[string]corerp.RecipePropertiesClassification {
-	recipes := map[string]map[string]corerp.RecipePropertiesClassification{}
-
-	// We are using the default recipe.
-	name := "default"
-
-	for _, repo := range repos {
-		// Skip dev environment recipes.
-		// dev repositories is in the form of ghcr.io/radius-project/dev/recipes/local-dev/secretstores:latest
-		// We should skip the dev repositories.
-		if isDevRepository(repo) {
+	validDevRecipes := map[string]map[string]corerp.RecipePropertiesClassification{}
+	for _, devRecipe := range availableDevRecipes() {
+		repo, err := remote.NewRepository(devRecipe.RepoPath)
+		if err != nil {
+			logger.Error(err, fmt.Sprintf("failed to create client to repository %s", devRecipe.RepoPath))
 			continue
 		}
 
-		resourceType := getResourceTypeFromPath(repo)
-		// If the resource type is empty, it means we don't support the repository.
-		if resourceType == "" {
+		// Check if the tag exists in the repository.
+		// Setting this is not the best way to go.
+		repo.TagListPageSize = 1000
+		tagExists := false
+		err = repo.Tags(ctx, "", func(tags []string) error {
```

Comment:
Is there a reason why we need to list the tags when we're looking for a specific one? It might be helpful to refactor this logic into a function. eg: `checkRefenceExists()`

Reply:
I updated this part and started using **FetchReference** instead of listing all the tags.